### PR TITLE
Wei method call defaults tc

### PR DESCRIPTION
**Pull request by danielytan**
Support for method calls, defaults, parameters

### CODE REVIEW

**Review by danielytan — COMMENTED** [review]
Looks good to me!

File: type-check.ts [hunk 1]
```diff
@@ -885,6 +889,25 @@ export function tcExpr(
               methodArgs.every((argTyp, i) => isAssignable(env, realArgs[i].a[0], argTyp))
             ) {
               return { ...expr, a: [methodRet, expr.a], obj: tObj, arguments: tArgs };
+            } else if (
```
Comment (xiexr151e) [inline type-check.ts]: I believe this is similar to how you would handle defaults with the `call` tag. This feels like duplicated code.

**Review by xiexr151e — COMMENTED** [review]
Left a comment for this code.

File: type-check.ts (hunk 1 above)
Comment (DavidLiu0304) [reply type-check.ts]: Hey Derek, thanks for your comment. I will look over it and try to move this logic to a function

**Review by xiexr151e — APPROVED** [review]
Nice and clean!

**Pull request merged**